include/circt/Dialect/FIRRTL/FIRRTLAnnotationHelper.h

```diff
@@ -84,6 +84,33 @@ struct AnnoTargetCache {
     return targets.lookup(name);
   }
 
+  /// Replace `oldOp` with `newOp` in the target cache. The new and old ops can
+  /// have different names.
+  void replaceOp(Operation *oldOp, Operation *newOp) {
+    auto namOp = dyn_cast<FNamableOp>(oldOp);
```

Comment:
Requiring these to be namable is out-of-sync with the code that populates this cache (`gatherTargets`).  A quick review shows the following are considered for annotation targets but would be ignored here: `MemoryPortOp` and `PrintFOp`.

Not sure if perhaps those should be `FNamable` or the "is nameable target considered for annotations" should be captured in a method somewhere, but it seems creation and updating should have same idea here.